plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt

```diff
@@ -24,6 +24,7 @@ import java.io.File
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.async
 import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.withContext
```

Comment:
> This relatively small deterioration is acceptable, because the parallelization now independently of the Yarn Berry version.

I'm not sure whether I can agree to that argumentation. Isn't it so that probably all newer Yarn Berry versions do have this built-in parallelization? And isn't it also so that it's more likely that people use a newer rather than an older version of Yarn Berry? And isn't it also better to rely on the tool's parallelization rather than re-implementing it?

The only plus that I see is the code alignment with other Node package managers. But that would really only weight-in if the code was also extracted to a common place.

Reply:
> I'm not sure whether I can agree to that argumentation. Isn't it so that probably all newer Yarn Berry versions do have this built-in parallelization? And isn't it also so that it's more likely that people use a newer rather than an older version of Yarn Berry? And isn't it also better to rely on the tool's parallelization rather than re-implementing it?

My main reasons why I believe this change to be good are: 1. solution is more general and 2. code alignment.

> The only plus that I see is the code alignment with other Node package managers. But that would really only weight-in if the code was also extracted to a common place.

Extracting common is is a topic I will address after cleaning up the Yarn2 impl.

Reply:
I ended up cleaning up the existing code instead of this rewrite, see: https://github.com/oss-review-toolkit/ort/pull/10233